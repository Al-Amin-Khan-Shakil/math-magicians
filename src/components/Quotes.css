.message {
  font-size: 2rem;
  color: #ffa340;
  margin: 60px;
}

.quote-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: 80vh;
}

.quote-title {
  font-size: 2.3rem;
  color: #0a3d62;
  width: 60vw;
  text-align: left;
}

.quote {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 20px;
  margin: 50px;
  width: 60vw;
}

.qtag {
  font-weight: 600;
  font-size: 1.3rem;
  color: #576574;
}

.author {
  align-self: flex-end;
  margin-right: 60px;
  font-weight: bolder;
  color: #0a3d62;
}
