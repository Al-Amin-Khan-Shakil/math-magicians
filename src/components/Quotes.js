import './Quotes.css';
import { useState, useEffect } from 'react';

const url = 'https://api.api-ninjas.com/v1/quotes?category=computers';
const apiKey = 'RCx4GyRkAnqwLwS2DBq/cg==6hZ3KyhXY6HyLkdm';

const Quotes = () => {
  const [quotes, setQuotes] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [isError, setIsError] = useState(false);

  const getData = async () => {
    try {
      setIsLoading(true);
      const response = await fetch(url, {
        method: 'GET',
        headers: {
          'X-Api-Key': apiKey,
          'Content-Type': 'application/json',
        },
      });
      const data = await response.json();
      setQuotes(data[0]);
      setIsLoading(false);
      if (!response.ok) {
        throw new Error('Network response was not ok');
      }
    } catch (error) {
      setIsError(true);
    }
  };

  useEffect(() => {
    getData();
  }, []);

  const message = isError ? 'There must be some error' : 'Loading...';
  if (isLoading || isError) {
    return <p className="message">{message}</p>;
  }
  return (
    <div className="quote-container">
      <h2 className="quote-title">Quote of the moment!</h2>
      <blockquote className="quote">
        <q className="qtag">{quotes.quote}</q>
        <cite className="author">
          ---
          {quotes.author}
        </cite>
      </blockquote>
    </div>
  );
};

export default Quotes;
