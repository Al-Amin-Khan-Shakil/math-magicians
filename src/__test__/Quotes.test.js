import renderer from 'react-test-renderer';
import Quotes from '../components/Quotes';

test('Test the Quotes component', () => {
  const tree = renderer.create(<Quotes />).toJSON();
  expect(tree).toMatchSnapshot();
});
