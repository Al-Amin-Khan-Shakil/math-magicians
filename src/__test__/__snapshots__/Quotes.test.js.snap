// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Quotes component 1`] = `
<div
  className="quote-container"
>
  <h2
    className="quote-title"
  >
    Quote of the moment!
  </h2>
  <blockquote
    className="quote"
  >
    <q
      className="qtag"
    />
    <cite
      className="author"
    >
      ---
    </cite>
  </blockquote>
</div>
`;
